File: backend/api/tm_routes/feedback.py
from __future__ import annotations

from fastapi import APIRouter

from backend.api.tm_routes.models import TermFeedbackPayload
from backend.services.learning_service import record_term_feedback

router = APIRouter()


@router.post("/feedback")
async def tm_record_feedback(payload: TermFeedbackPayload) -> dict:
    """
    Record user feedback for terminology learning.
    """
    try:
        record_term_feedback(
            payload.source, payload.target, payload.source_lang, payload.target_lang
        )
        return {"status": "recorded"}
    except Exception as exc:
        import logging

        logging.getLogger(__name__).error(f"Feedback recording failed: {exc}")
        return {"status": "error", "message": str(exc)}

File: frontend/src/hooks/terminology/feedback.js
import { tmApi } from '../../services/api/tm';

export const recordFeedbackWithDebounce = ({
    feedbackTimerRef,
    source,
    target,
    sourceLang,
    targetLang,
}) => {
    if (!source || !target) return;
    if (feedbackTimerRef.current) clearTimeout(feedbackTimerRef.current);
    feedbackTimerRef.current = setTimeout(async () => {
        try {
            await tmApi.feedback({
                source,
                target,
                source_lang: sourceLang,
                target_lang: targetLang,
            });
        } catch (error) {
            console.error('Failed to record feedback:', error);
        }
    }, 2000);
};
